presto-hive/src/main/java/io/prestosql/plugin/hive/HiveType.java

```diff
@@ -238,17 +240,13 @@ private static TypeSignature getTypeSignature(TypeInfo typeInfo)
                 return primitiveType.getTypeSignature();
             case MAP:
                 MapTypeInfo mapTypeInfo = (MapTypeInfo) typeInfo;
-                TypeSignature keyType = getTypeSignature(mapTypeInfo.getMapKeyTypeInfo());
-                TypeSignature valueType = getTypeSignature(mapTypeInfo.getMapValueTypeInfo());
-                return new TypeSignature(
-                        StandardTypes.MAP,
-                        ImmutableList.of(TypeSignatureParameter.typeParameter(keyType), TypeSignatureParameter.typeParameter(valueType)));
+                return mapType(
```

Comment:
"Simplify HiveType.getTypeSignature with factory methods" also looks good % one comment.
You can split it out together with "Use the passed type to serialize in SerDeUtils"